core/drivers/wdt/watchdog_sm.c

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022 Microchip
+ */
+
+#include <drivers/wdt.h>
+#include <sm/optee_smc.h>
+#include <sm/psci.h>
+
+enum smcwd_call {
+	SMCWD_INIT		= 0,
+	SMCWD_SET_TIMEOUT	= 1,
+	SMCWD_ENABLE		= 2,
+	SMCWD_PET		= 3,
+	SMCWD_GET_TIMELEFT	= 4,
+};
+
+static unsigned long wdt_min_timeout;
+static unsigned long wdt_max_timeout;
+
+enum sm_handler_ret wdt_sm_handler(struct thread_smc_args *args)
+{
+	if (args->a0 != CFG_WDT_SM_HANDLER_ID)
```

Comment:
To better optimize the code in `sm_from_nsec()` we could do this test in an inline function which would then be executed as part of `sm_from_nsec()` avoiding the jump to this function only to return most often. Do you get the idea?

Reply:
Yes totally. This would avoid a function call each time we do the test, acked.